docs/examples/ex39.py

```diff
@@ -0,0 +1,110 @@
+r"""Heat equation in one-dimension.
+
+The solution of the heat equation is reduced from two dimensions in ex19 to
+one here, to demonstrate the different post-processing required.
+
+The two-dimensional modes from ex19 reduce in the limit
+:math:`w_1 \rightarrow\infty` of the strip :math:`|x| < w` to
+
+.. math::
+    \exp \left[
+      -\kappa
+      \left\{
+        \left(\frac{(2n + 1)\pi}{2w}\right)^2
+      \right\}
+      t
+    \right]
+    \cos\frac{(2n + 1)\pi x}{2w}
+for :math:`n = 0, 1, 2, \ldots`.
+
+Here we simulate the decay of the fundamental, :math:`n = 0`, again
+discretizing time using the generalized ('theta method') trapezoidal
+rule.
+
+"""
+from typing import Iterator, Tuple
+
+import numpy as np
+from scipy.sparse.linalg import splu
+
+from skfem import *
+from skfem.models.poisson import laplace, mass
+
+
+halfwidth = 2.0
+ncells = 2 ** 3
+diffusivity = 5.0
+
+mesh = MeshLine(np.linspace(-1, 1, 2 * ncells) * halfwidth)
+
+element = ElementLineP2()  # or ElementLineP1
+basis = Basis(mesh, element)
+
+L = diffusivity * asm(laplace, basis)
+M = asm(mass, basis)
+
+dt = 0.01
+print("dt =", dt)
+theta = 0.5  # Crank–Nicolson
+L0, M0 = penalize(L, M, D=basis.find_dofs())
+A = M0 + theta * L0 * dt
+B = M0 - (1 - theta) * L0 * dt
+
+backsolve = splu(A.T).solve  # .T as splu prefers CSC
+
+u_init = np.cos(np.pi * basis.doflocs[0] / 2 / halfwidth)
+
+
+def evolve(t: float, u: np.ndarray) -> Iterator[Tuple[float, np.ndarray]]:
```

Comment:
Can you think about out how this could be run in the test suite? In particular, `tests/test_examples.py` should ideally run all examples so that they won't accidentally break. Maybe add some call to `evolve` to `tests/test_examples.py`?

Reply:
I've launched #678 for this.